src/entitysdk/util.py

```diff
@@ -0,0 +1,47 @@
+"""Utility functions."""
+
+from json import dumps
+
+import httpx
+
+from entitysdk.common import ProjectContext
+
+
+def make_db_api_request(
+    url: str,
+    *,
+    method: str,
+    json: dict | None = None,
+    parameters: dict | None = None,
+    project_context: ProjectContext,
+    token: str,
+    http_client: httpx.Client | None = None,
+) -> httpx.Response:
+    """Make a request to entitycore api."""
+    if http_client is None:
+        http_client = httpx.Client()
+
+    response = http_client.request(
+        method=method,
+        url=url,
+        headers={
+            "project-id": project_context.project_id,
+            "virtual-lab-id": project_context.virtual_lab_id,
+            "Authorization": f"Bearer {token}",
+        },
+        json=json,
+        params=parameters,
+    )
+
+    try:
+        response.raise_for_status()
+    except httpx.HTTPError as e:
+        message = (
+            f"{method} {url}\n"
+            f"json : {dumps(json, indent=2)}\n"
+            f"params : {parameters}\n"
+            f"response : {response.text}"
+        )
+        raise httpx.HTTPError(message=message) from e
```

Comment:
If we intercept the error, I think that it would be better to raise a custom exception (from e), or we lose access to the  `.request` and `.response` properties in case we need them (and it's not clear that you should access the nested exception to get the original request or response).

Note also that `httpx.RequestError` can be raised when calling `.request()`, but we are not checking that.

Reply:
Yes, did that and captured the RequestError too.